libc/include/assert.h.def

```diff
@@ -12,4 +12,21 @@
 // This file may be usefully included multiple times to change assert()'s
 // definition based on NDEBUG.
 
+#ifndef __cplusplus
```

Comment:
if you're adding this here it needs to be removed from the `api.td` files it's in (specifically baremetal, linux, and GPU).